instrumentation/github.com/tidwall/buntdb/splunkbuntdb/test/buntdb_test.go

```diff
@@ -0,0 +1,469 @@
+	testView(t, "DescendRange", func(tx *splunkbuntdb.Tx) error {
+		var arr []string
+		err := tx.DescendRange("test-index", "4", "2", func(key, value string) bool {
+			arr = append(arr, key, value)
+			return true
+		})
+		assert.NoError(t, err)
+		assert.Equal(t, []string{
+			"regular:d", "4",
+			"regular:c", "3",
+		}, arr)
+		return nil
+	})
+}
+
+func TestDropIndex(t *testing.T) {
+	testUpdate(t, "DropIndex", func(tx *splunkbuntdb.Tx) error {
+		err := tx.DropIndex("test-index")
+		assert.NoError(t, err)
+		return nil
+	})
+}
+
+func TestGet(t *testing.T) {
+	testView(t, "Get", func(tx *splunkbuntdb.Tx) error {
+		val, err := tx.Get("regular:a")
+		assert.NoError(t, err)
+		assert.Equal(t, "1", val)
+		return nil
+	})
+}
+
+func TestIndexes(t *testing.T) {
+	testView(t, "Indexes", func(tx *splunkbuntdb.Tx) error {
+		indexes, err := tx.Indexes()
+		assert.NoError(t, err)
+		assert.Equal(t, []string{"test-index", "test-spatial-index"}, indexes)
+		return nil
+	})
+}
+
+func TestIntersects(t *testing.T) {
+	testView(t, "Intersects", func(tx *splunkbuntdb.Tx) error {
+		var arr []string
+		err := tx.Intersects("test-spatial-index", "[3 3],[4 4]", func(key, value string) bool {
+			arr = append(arr, key, value)
+			return true
+		})
+		assert.NoError(t, err)
+		assert.Equal(t, []string{
+			"spatial:c", "[3 3]",
+			"spatial:d", "[4 4]",
+		}, arr)
+		return nil
+	})
+}
+
+func TestLen(t *testing.T) {
+	testView(t, "Len", func(tx *splunkbuntdb.Tx) error {
+		n, err := tx.Len()
+		assert.NoError(t, err)
+		assert.Equal(t, 10, n)
+		return nil
+	})
+}
+
+func TestNearby(t *testing.T) {
+	testView(t, "Nearby", func(tx *splunkbuntdb.Tx) error {
+		var arr []string
+		err := tx.Nearby("test-spatial-index", "[3 3]", func(key, value string, distance float64) bool {
+			arr = append(arr, key, value)
+			return false
+		})
+		assert.NoError(t, err)
+		assert.Equal(t, []string{
+			"spatial:c", "[3 3]",
+		}, arr)
+		return nil
+	})
+}
+
+func TestSet(t *testing.T) {
+	testUpdate(t, "Set", func(tx *splunkbuntdb.Tx) error {
+		previousValue, replaced, err := tx.Set("regular:a", "11", nil)
+		assert.NoError(t, err)
+		assert.True(t, replaced)
+		assert.Equal(t, "1", previousValue)
+		return nil
+	})
+}
+
+func TestTTL(t *testing.T) {
+	testUpdate(t, "TTL", func(tx *splunkbuntdb.Tx) error {
+		duration, err := tx.TTL("regular:a")
+		assert.NoError(t, err)
+		assert.Equal(t, time.Duration(-1), duration)
+		return nil
+	})
+}
+
+func withTestingDeadline(ctx context.Context, t *testing.T) context.Context {
+	d, ok := t.Deadline()
+	if !ok {
+		d = time.Now().Add(10 * time.Second)
+	} else {
+		d = d.Add(-time.Millisecond)
+	}
+	var cancel context.CancelFunc
+	ctx, cancel = context.WithDeadline(ctx, d)
+	t.Cleanup(cancel)
+	return ctx
+}
+
+func testUpdate(t *testing.T, name string, f func(tx *splunkbuntdb.Tx) error) {
+	sr := tracetest.NewSpanRecorder()
+	tp := trace.NewTracerProvider(trace.WithSpanProcessor(sr))
+
+	global.SetTracerProvider(tp)
```

Comment:
Instead of setting a global `TracerProvider` which can have side effects, please instead explicitly pass the `TracerProvider` with the `WithTracerProvider` option?